internal/middleware/authorize.go

```diff
@@ -0,0 +1,62 @@
+package middleware
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/fdanis/yg-loyalsys/internal/common"
+	"github.com/golang-jwt/jwt/v4"
+)
+
+type AuthorizeMiddleware struct {
+	secretKey string
+}
+
+func NewAuthorizeMiddleware(key string) (a *AuthorizeMiddleware) {
+	return &AuthorizeMiddleware{secretKey: key}
+}
+
+func (a *AuthorizeMiddleware) Authorize(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		authHeader := r.Header.Get("Authorization")
+		if authHeader == "" {
+			unauthorized(w)
+			return
+		} else {
```

Comment:
тут else не нужен. потому что сверху в ифе return
```suggestion
		} 
```

Reply:
исправил